### PR TITLE
Add RISC-V Vector support.

### PR DESCRIPTION
**Pull request by luhenry**
Hello,

This adds support for the RISC-V architecture by using the RISC-V Vector Intrinsics [1]. I've added support for CI similarly to ARM SVE.

Thank you for your review!

Fixes https://github.com/xtensor-stack/xsimd/issues/503

[1] https://github.com/riscv-non-isa/rvv-intrinsic-doc

### CODE REVIEW

**Comment by JohanMabille** [comment]
Thanks a lot for your PR, this is huge! I'll take the time to review it in detail over the weekend.

**Comment by SylvainCorlay** [comment]
Wow. Thanks a lot for that. 🙇

**Comment by luhenry** [comment]
There is a test failure I can't reproduce locally. One of the difference is the QEMU version so I'll debug that out next week.

File: include/xsimd/arch/xsimd_rvv.hpp [hunk 1]
```diff
@@ -0,0 +1,1451 @@
...
+#define RVV_IDENTIFIER9(T,S,M, t, ...) t
+#define RVV_IDENTIFIER8(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER9(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER7(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER8(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER6(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER7(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER5(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER6(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER4(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER5(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER3(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER4(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER2(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER3(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER1(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER2(T,S,M, __VA_ARGS__)))
+#define RVV_IDENTIFIER0(T,S,M, t, p, ...) RVV_JOIN(t, RVV_PREFIX##p(T,S,M, RVV_IDENTIFIER1(T,S,M, __VA_ARGS__)))
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_rvv.hpp:51]: either undef all these at the end of the header, or prefix them with `XSIMD_` ?

File: include/xsimd/arch/xsimd_rvv.hpp [hunk 2]
```diff
@@ -0,0 +1,1451 @@
...
+            /* TODO: mention only relevant types */ \
+            NAME##_cruft::impl_t<  int8_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t< uint8_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t< int16_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t<uint16_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t< int32_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t<uint32_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t< int64_t,RVV_M_VALUE>, \
+            NAME##_cruft::impl_t<uint64_t,RVV_M_VALUE>, \
+            IF_zvfh(NAME##_cruft::impl_t<_Float16,RVV_M_VALUE>,) \
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_rvv.hpp:194]: We don't support half precision float in xsimd yet

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**